src/main/java/org/terasology/dialogs/action/NewDialogActionTypeHandler.java

```diff
@@ -17,33 +17,33 @@
 package org.terasology.dialogs.action;
 
 import java.util.Map;
+import java.util.Optional;
 
-import org.terasology.persistence.typeHandling.DeserializationContext;
-import org.terasology.persistence.typeHandling.PersistedData;
-import org.terasology.persistence.typeHandling.PersistedDataMap;
-import org.terasology.persistence.typeHandling.RegisterTypeHandler;
-import org.terasology.persistence.typeHandling.SerializationContext;
-import org.terasology.persistence.typeHandling.SimpleTypeHandler;
+import org.terasology.persistence.typeHandling.*;
 
 import com.google.common.collect.ImmutableMap;
 
 @RegisterTypeHandler
-public class NewDialogActionTypeHandler extends SimpleTypeHandler<NewDialogAction> {
+public class NewDialogActionTypeHandler extends TypeHandler<NewDialogAction> {
 
     @Override
-    public PersistedData serialize(NewDialogAction action, SerializationContext context) {
+    public PersistedData serialize(NewDialogAction action, PersistedDataSerializer context) {
         Map<String, PersistedData> data = ImmutableMap.of(
-                "type", context.create(action.getClass().getSimpleName()),
-                "target", context.create(action.getTarget()));
+                "type", context.serialize(action.getClass().getSimpleName()),
+                "target", context.serialize(action.getTarget()));
```

Comment:
Same as above -- `"type"` mapping is unnecessary, you could simply `return context.serialize(action.getTarget())`